### PR TITLE
Changes related to "Modify Program DTO so all Program REST endpoints include some Data Center Information"

### PR DESCRIPTION
**Pull request by Azher2Ali**
In this ticket #420,we have modified and added more functionality to the Program Service to serve as a Data Center Registry for ARGO's Regional Data Processing Centers (RDPCs). Each program now belongs to a single Data Center, and relevant Data Center details are included in the Program entity. These changes impact five endpoints: CreateProgram, UpdateProgram, ListPrograms, GetProgram, and List Data Center Programs.

The screenshot below depicts that "List Data Center Programs" now includes a list of programs where each program should contain the specified information.
List Data Center Programs

### CODE REVIEW

**Comment by joneubank** [comment]
Strong descriptions of the PRs, very appreciated and helpful. Would love if the PR titles were of matching quality so we can recognize the PR at a glance.

File: src/main/java/org/icgc/argo/program_service/services/ProgramServiceFacade.java
Comment (joneubank) [inline src/main/java/org/icgc/argo/program_service/services/ProgramServiceFacade.java:270]: Good place for builder pattern?

File: src/main/java/org/icgc/argo/program_service/converter/Grpc2JsonConverter.java
Comment (joneubank) [inline src/main/java/org/icgc/argo/program_service/converter/Grpc2JsonConverter.java:144]: I wrote a comment here initially asking why we are suppressing so many errors... but of course this is in grpc code that we are deprecating. Can you add a code comment calling this out to explain why we are happy to suppress errors here?

**Review by joneubank — COMMENTED** [review]
This set of additions looks good. There is a lot of code in this PR which comes from PRs with code changes requested, so I will only mark this review with a comment instead of approval so we don't merge in those changes that need some updates. Once the other PRs are addressed we can approve this work.

File: src/main/java/org/icgc/argo/program_service/converter/Grpc2JsonConverter.java
Comment (Azher2Ali) [reply src/main/java/org/icgc/argo/program_service/converter/Grpc2JsonConverter.java]: Handled all exceptions as ProgramRuntimeException, throwing to the caller with the error message

File: src/main/java/org/icgc/argo/program_service/model/dto/DataCenterDetailsDTO.java
Comment (justincorrigible) [inline src/main/java/org/icgc/argo/program_service/model/dto/DataCenterDetailsDTO.java:2]:
```suggestion
 * Copyright (c) 2023 The Ontario Institute for Cancer Research. All rights reserved

```

File: src/main/java/org/icgc/argo/program_service/services/ProgramService.java
Comment (justincorrigible) [inline src/main/java/org/icgc/argo/program_service/services/ProgramService.java:475]: should this be @NonNull as well?

**Comment by justincorrigible** [comment]
> Once the other PRs are addressed we can approve this work.
Looks to me like we're at that point @joneubank 
Thoughts?

**Comment by joneubank** [comment]
Looks good. If it builds we can merge.

**Review by joneubank — APPROVED** [review]

**Pull request merged**
